presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/decoders/TimestampDecoder.java

```diff
@@ -43,22 +85,48 @@ public TimestampDecoder(ConnectorSession session, String path)
         this.zoneId = ZoneId.of(session.getTimeZoneKey().getId());
     }
 
+    private static LocalDateTime query(TemporalAccessor parsedDateTime)
+    {
+        try {
+            return LocalDateTime.from(parsedDateTime);
+        }
+        catch (DateTimeException ex) {
+        }
+
+        try {
+            LocalDate date = LocalDate.from(parsedDateTime);
+            return date.atStartOfDay();
+        }
+        catch (DateTimeException ex) {
+            throw new PrestoException(NOT_SUPPORTED, "Unable to obtain LocalDateTime from TemporalAccessor: " +
+                    parsedDateTime + " of type " + parsedDateTime.getClass().getName(), ex);
+        }
+    }
+
     @Override
     public void decode(SearchHit hit, Supplier<Object> getter, BlockBuilder output)
     {
-        DocumentField documentField = hit.getFields().get(path);
+        Map<String, Object> documentMap = hit.getSourceAsMap();
```

Comment:
The hit.fields is not populated when we query es from a nested data structure. I believe this is due to the fact that nested objects internally exist on a different lucene index. So to address this we need to pull the raw source that gets returned.

Reply:
We may want to try with hit.getFields() first and fall back to the other one if that's missing. In any case, we shouldn't have to traverse the source document again. `getter.get()` should provide the value corresponding to the field we're interested in (see the other decoders for examples)